soc/arm/nordic_nrf/soc_nrf_common.h

```diff
@@ -66,14 +66,21 @@
  * @param prop_name human-readable string name for 'prop'
  */
 #define NRF_DT_CHECK_GPIO_CTLR_IS_SOC(node_id, prop, prop_name)		\
-	BUILD_ASSERT(!DT_NODE_HAS_PROP(node_id, prop) ||		\
-		     DT_NODE_HAS_COMPAT(DT_GPIO_CTLR(node_id, prop),	\
-					nordic_nrf_gpio),		\
-		     "Devicetree node " DT_NODE_PATH(node_id)		\
-		     " property " prop_name " must refer to a GPIO "	\
-		     " controller with compatible nordic,nrf-gpio; "	\
-		     "got " DT_NODE_PATH(DT_GPIO_CTLR(node_id, prop))	\
-		     ", which does not have this compatible")
```

Comment:
> soc: nrf: fix NRF_DT_CHECK_GPIO_CTLR_IS_SOC
> This was merged by mistake without being tested and is not working properly.

Does this commit need to be in 2.5?

Reply:
> Does this commit need to be in 2.5?

Marking resolved since that particular commit has its own PR now, which can be merged into 2.5 if needed.